gui/mozregui/addons_editor.py

```diff
@@ -0,0 +1,89 @@
+from PyQt4.QtCore import QAbstractListModel, QModelIndex, Qt, \
+    pyqtSlot as Slot
+from PyQt4.QtGui import QWidget, QFileDialog
+from mozregui.ui.addons_editor import Ui_AddonsEditor
+
+
+class AddonsMode(QAbstractListModel):
+    """
+    A Qt model that can edit addons path.
+    """
+
+    def __init__(self, parent=None):
+        QAbstractListModel.__init__(self, parent)
+        self.addonsPath = []
+
+    def rowCount(self, index=QModelIndex()):
+        return len(self.addonsPath)
+
+    def data(self, index, role=Qt.DisplayRole):
+        if role == Qt.DisplayRole:
+            if index.row() == 0:
+                return None
+            else:
+                return repr(self.addonsPath[index.row()])
+
+    def setData(self, index, new_value, role=Qt.EditRole):
+        path = str(new_value.toString())
+        self.addonsPath[index.row()] = path
+        return True
+
+    def headerData(self, section, orientation, role=Qt.DisplayRole):
+        if role == Qt.DisplayRole and orientation == Qt.Horizontal:
+            return 'path'[section]
+
+    def flags(self, index):
+        return Qt.ItemIsSelectable | Qt.ItemIsEditable | Qt.ItemIsEnabled
+
+    def add_addons_from_file(self, addons):
+        if addons:
+            path = len(self.addonsPath)
+            self.beginInsertRows(QModelIndex(), path,
+                                 path + len(addons) - 1)
+            self.addonsPath.extend(str(addons))
```

Comment:
After I insert some path like "/User/mikeling/mozregression/README.md", it will become something like ['/', 'U', 's', 'e', 'r', 's', '/', 'm', 'i', 'k', 'e', 'l', 'i', 'n', 'g', '/', 'm', 'o', 'z', 'r', 'e', 'g', 'r', 'e', 's', 's', 'i', 'o', 'n', '/', 'R', 'E', 'A', 'D', 'M', 'E', '.', 'm', 'd'] in self.addonsPath. And I don't know where goes wrong. 

I guess it's wrong because I define a wrong setData method. But, actually, I haven't fully understand setData method even after read the document about it. Could you help me about that? Thank you very much! @parkouss